processor/src/main/java/org/mobilitydata/gtfsvalidator/processor/LatLonValidatorGenerator.java

```diff
@@ -0,0 +1,154 @@
+ * detected automatically.
+ */
+public class LatLonValidatorGenerator {
+
+  private static final String VALIDATOR_PACKAGE_NAME = "org.mobilitydata.gtfsvalidator.validator";
+
+  private final List<GtfsFileDescriptor> fileDescriptors;
+
+  public LatLonValidatorGenerator(List<GtfsFileDescriptor> fileDescriptors) {
+    this.fileDescriptors = fileDescriptors;
+  }
+
+  public List<JavaFile> generateValidatorFiles() {
+    List<JavaFile> validators = new ArrayList<>();
+    for (GtfsFileDescriptor fileDescriptor : fileDescriptors) {
+      if (!fileDescriptor.latLonFields().isEmpty()) {
+        validators.add(generateValidator(fileDescriptor));
+      }
+    }
+    return validators;
+  }
+
+  private static JavaFile generateValidator(GtfsFileDescriptor fileDescriptor) {
+    GtfsEntityClasses entityClasses = new GtfsEntityClasses(fileDescriptor);
+    TypeSpec.Builder typeSpec =
+        TypeSpec.classBuilder(validatorName(fileDescriptor))
+            .addModifiers(Modifier.PUBLIC, Modifier.FINAL)
+            .addAnnotation(Generated.class)
+            .addAnnotation(GtfsValidator.class)
+            .superclass(
+                ParameterizedTypeName.get(
+                    ClassName.get(SingleEntityValidator.class),
+                    entityClasses.entityImplementationTypeName()));
+
+    addPoles(typeSpec);
+
+    MethodSpec.Builder validateMethod =
+        MethodSpec.methodBuilder("validate")
+            .addModifiers(Modifier.PUBLIC)
+            .addAnnotation(Override.class)
+            .returns(void.class)
+            .addParameter(entityClasses.entityImplementationTypeName(), "entity")
+            .addParameter(NoticeContainer.class, "noticeContainer");
+
+    for (LatLonDescriptor latLonDescriptor : fileDescriptor.latLonFields()) {
+      validateLatLon(fileDescriptor, latLonDescriptor, validateMethod);
+    }
+
+    typeSpec.addMethod(validateMethod.build());
+
+    return JavaFile.builder(VALIDATOR_PACKAGE_NAME, typeSpec.build()).build();
+  }
+
+  private static void addPoles(TypeSpec.Builder typeSpec) {
+    typeSpec.addField(
+        FieldSpec.builder(
+                S2LatLng.class, "NORTH_POLE", Modifier.PRIVATE, Modifier.STATIC, Modifier.FINAL)
+            .initializer("$T.fromDegrees(90, 0)", S2LatLng.class)
+            .build());
+    typeSpec.addField(
+        FieldSpec.builder(
+                S2LatLng.class, "SOUTH_POLE", Modifier.PRIVATE, Modifier.STATIC, Modifier.FINAL)
+            .initializer("$T.fromDegrees(-90, 0)", S2LatLng.class)
+            .build());
+  }
+
+  private static void validateLatLon(
+      GtfsFileDescriptor fileDescriptor,
+      LatLonDescriptor latLonDescriptor,
+      MethodSpec.Builder validateMethod) {
+    validateMethod.beginControlFlow(
+        "if (entity.$L())", FieldNameConverter.hasMethodName(latLonDescriptor.latLonField()));
+
+    validateMethod
+        .beginControlFlow(
+            "if (Math.abs(entity.$L()) <= 1.0 && Math.abs(entity.$L()) <= 1.0)",
```

Comment:
It would be very nice to explain why `1.0` is chosen :)

Reply:
I don't think so. This is just a small square (-1, -1) - (1, 1) around the origin.